source/unified-test-format/unified-test-format.md

```diff
@@ -923,6 +923,18 @@ The structure of this object is as follows:
   to have. The test runner MUST assert that the error does not contain any of the specified labels (e.g. using the
   `hasErrorLabel` method).
 
+- `writeErrors`: Optional document. The write errors expected to be present in the error. The `writeErrors` document
+  contains numeric keys representing the index of the write that failed and `writeError` object values. The test runner MUST
+  assert that the error contains a `writeError` for each index present in `writeErrors` and MUST assert that the `writeError`s
+  match as root-level documents according to the rules in [Evaluating Matches](#evaluating-matches). The test runner MUST
+  assert that the error does not contain any additional `writeError`s.
+
+- `writeConcernErrors`: Optional array of one or more objects. An ordered list of write concern errors expected to be
+  present in the error. The test runner MUST assert that each `writeConcernError` in this list matches the
+  `writeConcernError` present at the same index in the error's list of `writeConcernError`s as a root-level document according
+  to the rules in [Evaluating Matches](#evaluating-matches). The test runner MUST assert that the error does not contain any
+  additional `writeConcernErrors`s.
```

Comment:
Do you suppose these fields are suitable for use with the collection-level bulkWrite method defined in the CRUD spec? There, BulkWriteException still has a `writeErrors` field. Although `writeConcernError` is singular, the unified spec could always suggest wrapping it in an array so that `expectedError` assertions can be performed consistently.

I realize this is outside the scope of this PR, but I'm just thinking more generally about whether these new fields are limited to the client API or could get more general use.

Reply:
I don't see why not. Your proposal for write concern errors seems reasonable. The collection-level `BulkWriteException` has a different format for its `writeErrors` field (a list of individual `BulkWriteError`s that store the index of the operation as a field) that doesn't fully align with the index-to-error mapping defined here, but drivers should still be able to make the right assertions.

Reply:
> The collection-level BulkWriteException has a different format for its writeErrors field

Ah, thanks for pointing that out. In that case, we'd need to sort out _both_ the `writeConcernError` (single to array) and `writeErrors` (object with `index` fields to a map) in order to share this structure.

In the meantime, can you clarify that both of these assertion fields are only intended to be used for the client API's BulkWriteException?

Reply:
yep, updated